llama-index-core/llama_index/core/evaluation/retrieval/metrics.py

```diff
@@ -12,45 +12,51 @@
 
 
 class HitRate(BaseRetrievalMetric):
-    """Hit rate metric."""
+    """Hit rate metric: Compute the proportion of matches between retrieved documents and expected documents."""
```

Comment:
I like the metric, but I'm not sure I like it as a replacement to HitRate, since "Hit" implies a binary nature.

Perhaps we can call this `MeanDocumentMatch` or something that represents an average of proportions.

Reply:
To be clear, I am recommending leaving `HitRate` as is and creating a net new `BaseRetrievalMetric`

Reply:
That’s a good point. For the proposed implementation, would you not say that _technically_ each document gets a binary representation in the score (since each doc is hit, or not hit)?

Then the summation and following division allows for the accounting, and comparison, of multiple docs in the score.

_Note_: I realise this would alter the usage pattern, compared to past HitRate implementations, so that adds a cost to this idea!

However, **in one metric**, we could get the binary representation **and** also avoid the information lost by having only 1 or 0 for scoring options, regardless of the amount of docs used.

Let me know what you think

Reply:
Yeah, each doc definitely gets assigned a binary value to it i.e. hit or no hit. But yea the usage pattern would be broken here unfortunately if we go down to that level of granularity with `HitRate`.

Perhaps your suggesting some sort of parameter to control the calculation of the `HitRate`, which if that's the case I think would be fine. We could make the default calculation be the current computation, and allow the user to specify if they would like to use the more granular calculation.

Reply:
Great idea! 

I'll work on that